### ISSUE TITLE
Problem when compiling with Visual Studio 2015 / 2017

### ISSUE THREAD

**Issue by ooer** [modelica-tools/FMUComplianceChecker#1]
Hi,

I get the following error, when I try to compile the Compliance Checker with Visual Studio 2015 or 2017
> fatal error C1189: #error:  Macro definition of vsnprintf conflicts with Standard Library function declaration 

The error is caused by the redefinition of (v)snprintf in the _c99_snprintf_ project.

I suggest to insert the following preprocessor directive.

```c
#if defined(_MSC_VER) && _MSC_VER >= 1900
#define HAVE_SNPRINTF 1
#define HAVE_VSNPRINTF 1
#endif
```

Best 
Olaf

**Comment by efredriksson-modelon** [comment]
Hi, this was solved in FMIL by https://trac.jmodelica.org/ticket/5277. However, the checker currently uses FMIL 2.0.2 which does not have this fix. Line 139 in https://github.com/modelica-tools/FMUComplianceChecker/blob/master/CMakeLists.txt.

**Comment by tbeu** [comment]
Can you please also push to https://github.com/modelon/FMILibrary?

**Comment by tbeu** [comment]
Any news on this?

**Comment by hlarsson-modelon** [comment]
@tbeu The FMILibrary mirror on GitHub is not actively maintained and will likely be taken down. Please refer to the JModelica SVN repository for the latest updates.

**Comment by tbeu** [comment]
Thanks for clarification. Found a git mirror at https://github.com/svn2github/FMILibrary.

**Comment by hlarsson-modelon** [comment]
This is fixed in the 2.0.4 release.
